src/routes/transactions/mappers/common/transaction.mapper.ts

```diff
@@ -0,0 +1,142 @@
+import { Injectable } from '@nestjs/common';
+import {
+  isEthereumTransaction,
+  isModuleTransaction,
+  isMultisigTransaction,
+  Transaction as TransactionDomain,
+} from '../../../../domain/safe/entities/transaction.entity';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { ModuleTransactionMapper } from '../module-transactions/module-transaction.mapper';
+import { MultisigTransactionMapper } from '../multisig-transactions/multisig-transaction.mapper';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import { EthereumTransaction } from '../../../../domain/safe/entities/ethereum-transaction.entity';
+import { groupBy } from 'lodash';
+import { IncomingTransferMapper } from '../transfers/transfer.mapper';
+import { TransactionItem } from '../../entities/transaction-item.entity';
+import { TransactionDomainGroup } from '../../entities/transaction-history-group.entity';
+import { DateLabel } from '../../entities/date-label.entity';
+
+@Injectable()
+export class TransactionMapper {
+  constructor(
+    private readonly multisigTransactionMapper: MultisigTransactionMapper,
+    private readonly moduleTransactionMapper: ModuleTransactionMapper,
+    private readonly incomingTransferMapper: IncomingTransferMapper,
+  ) {}
+  async mapTransaction(
+    chainId: string,
+    transactionsDomain: TransactionDomain[],
+    safe: Safe,
+    timezoneOffset?: string,
+    offset?: number,
+  ): Promise<[TransactionItem | DateLabel]> {
+    const transactionList: any[] = [];
+
+    let prev_page_timestamp = 0;
+    if (offset !== undefined && offset > 0) {
+      // Get previous page label
+      const timestamp =
+        transactionsDomain[0].executionDate?.getTime() ??
+        (
+          transactionsDomain[0] as MultisigTransaction
+        ).submissionDate?.getTime();
+      if (timestamp !== undefined) {
+        prev_page_timestamp = this.getDayInMillis(timestamp, timezoneOffset);
+      } else {
+        throw Error('ExecutionDate cannot be null');
+      }
+      transactionsDomain = transactionsDomain.slice(1);
+    }
```

Comment:
Honestly, I find this section quite confusing.

The conditional branches should have some documentation to follow along. The nesting also doesn't help in my opinion.

It seems that we are trying to get the previous page timestamp therefore I'd probably encapsulate this in a function (suggestion).

Reply:
I'll move it to a function.

Reply:
Done 3ed0214c5e8c5f8593e35c3295694f62ae20da5e